pkg/operator/config.go

```diff
@@ -82,6 +95,24 @@ func getProviderControllerFromImages(platform configv1.PlatformType, images Imag
 	}
 }
 
+// When the platform type is BareMetal, MAO is also responsible for deploying
+// the baremetal-operator pod
+func getAdditionalProviderImages(platform configv1.PlatformType, images Images) ([]string, error) {
+        noImages := []string{"", "", ""}
+	switch platform {
+	case configv1.BareMetalPlatformType:
+                if (images.BaremetalOperator == "" || images.BaremetalProvider == "") {
```

Comment:
This should be made more generic; we should have a list of known-image fields somewhere for this and loop through here.

Reply:
I realize now that this check is unnecessary.